CMakeLists.txt

```diff
@@ -22,6 +22,18 @@ endif ()
 
 set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
 
+
+# Enable a higher warning level and warnings-as-errors
+include (MongoC-Warnings)
+option (ENABLE_MORE_WARNINGS_AS_ERRORS "Enable extra warnings-as-errors in the build." ON)
+if (ENABLE_MORE_WARNINGS_AS_ERRORS)
+   mongoc_add_platform_compile_options (
+      msvc:/W3 msvc:/WX msvc:/wd4996
+      gnu-like:-Werror
+      gnu-like:-Wno-missing-braces
```

Comment:
Does this pertain to [MONGOCRYPT-389](https://jira.mongodb.org/browse/MONGOCRYPT-389)? If so, should that issue (currently still in "Needs Triage" status) also be referenced in this PR and advanced?

Also, I'll point out that [MONGOCRYPT-390](https://jira.mongodb.org/browse/MONGOCRYPT-390) is currently only mentioned in the PR description. You may want to add it to either the PR title or a commit message unless you indent to take care of that at merge time.